File: ArkhamAsylum.Lib/Models/Floor.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace ArkhamAsylum.Lib.Models
{
    public class Floor : Entity
    {
        public int Number { get; set; }

        public Guid BuildingId { get; set; }

        [JsonIgnore]
        public Building Building { get; set; }

        [JsonIgnore]
        public ICollection<Room> Rooms { get; set; }
    }
}

File: ArkhamAsylum.Web/wwwroot/Scripts/Models/Floor.js
class Floor extends Entity
{
    constructor(json)
    {
        super(json);
        if (json)
        {
            this.Number = json.number;
            this.BuildingId = json.buildingId;
        }
        else
        {
            this.Number = "";
            this.BuildingId = "00000000-0000-0000-0000-000000000000";
        }
    }
}